aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -213,6 +213,12 @@ static std::tuple<Tensor &,Tensor &> max_out_impl(Tensor& max, Tensor& max_indic
               "max only supports CPU AND CUDA device type, got: ", self.device().type());
   TORCH_CHECK(self.layout() == Layout::Strided,
               "max only supports strided layout, got: ", self.layout());
+  TORCH_CHECK(self.device().type() == max.device().type(),
+              "expected device type ", self.device().type(), " but got ",
```

Comment:
You should just check that the `device()` are equal, as mixed-device operations won't work on this kernel anyway